lib/cacheWrapper.js

```diff
@@ -1,76 +1,116 @@
 'use strict';
 
 var cacheWrapper = module.exports = {};
-
-var _ = require( 'lodash' );
-var Catbox = require( 'catbox' );
-var catboxMemory = require( 'catbox-memory' );
-var catboxRedis = require( 'catbox-redis' );
-var Q = require( 'q' );
-
+var _ = require('lodash');
+var Catbox = require('catbox');
+var catboxRedis = require('catbox-redis');
+var redisWrapper = require('./redisWrapper');
+var Q = require('q');
 var cacheClient;
 var startingClient = false;
 var queryQueue = [];
 var policies = [];
+var partition = '';
+
+cacheWrapper.initialise = function(serverConfig, cachePolicies) {
+  // no config, no cache...
+  if (_.isEmpty(serverConfig) || _.isEmpty(serverConfig.partition)) {
+    return Q.reject('server config missing partition');
+  }
+  if (_.isEmpty(cachePolicies)) {
+    return Q.reject('cachePolicies missing options(segment/expiresIn)');
+  }
+  partition = serverConfig.partition;
 
-cacheWrapper.initialise = function( serverConfig, cachePolicies ) {
   var deferred = Q.defer();
-  // if serverConfig is empty use an in-memory version of the cache
-  var catboxType = _.isEmpty( serverConfig ) ? catboxMemory : catboxRedis;
-  cacheClient = new Catbox.Client( catboxType, serverConfig );
 
+  cacheClient = new Catbox.Client(catboxRedis, serverConfig);
   // Build all of the cache policies into the client
-  _.each( cachePolicies, function( cachePolicy ) {
-    policies[cachePolicy.segment] = new Catbox.Policy( { expiresIn: cachePolicy.expiresIn }, cacheClient, cachePolicy.segment );
-  } );
+  _.each(cachePolicies, function(cachePolicy) {
+    if (!_.isUndefined(cachePolicy.segment) && !_.isUndefined(cachePolicy.expiresIn)) {
+      policies[cachePolicy.segment] = new Catbox.Policy(
+        {
+          expiresIn: cachePolicy.expiresIn
+        },
+        cacheClient, cachePolicy.segment
+      );
+    }
+  });
 
   // fire up the client
-  cacheClient.start( function() {
-    deferred.resolve();
-  } );
+  cacheClient.start(function() {
+    redisWrapper.initialise(serverConfig)
+    .then(function() {
+      deferred.resolve();
+    })
+    .fail(function() {
+      deferred.resolve('redis client was not initialised');
+    });
+  });
   return deferred.promise;
 };
 
 /**
 * Finds a value for the given key in the given segment of the memory store.
 * @param {Object} deferred - A promise to manage flow.
 * @param {Object} options - Including segment and key.
-* @returns {null} Doesn't explicitly return, simply handles the promis passed in.
+* @returns {null} Doesn't explicitly return, simply handles the promise passed in.
 */
-cacheWrapper._retrieve = function( deferred, options ) {
-  if ( policies[options.segment] ) {
-    policies[options.segment].get( options.key, function( error, cached ) {
-      if ( error || !cached ) {
+cacheWrapper._retrieve = function(deferred, options) {
```

Comment:
@notABluesSinger I spent a good few days reworking things so we didn't have to pass round deferreds but couldn't get there. The problem we have is because we queue things when the cache is unavailable, as we decouple the request from the response early, we need to create the deferred here and pass it down. Interested to hear any ideas though.